### PR TITLE
Shadow dagger

### LINKED ISSUE #905: Pulling conflicting guava versions (18.0 and 19.0)
We recently bumped `dagger` to 2.4 (#878), which pulls in guava 19 (Previously, dagger 2.0.2 pulled guava 18; in `versions.gradle` we specify guava 18).  After the dependency bump, everything builds in gradlew but we get build errors in IntelliJ. 

The specific error I saw is: `Error:java: java.lang.NoSuchMethodError: com.google.common.collect.ImmutableSetMultimap$Builder.putAll(Ljava/lang/Iterable;)Lcom/google/common/collect/ImmutableSetMultimap$Builder;`

Should we:

1. Use guava 19 (may need to force dependencies, at least http-remoting uses guava 18 but possibly there are others too)
2. Try to shadow guava
3. Something else

(Wanted to have a quick chat before running off and forcing dependency versions, as I may not fully understand the implications)

### PR DESCRIPTION
**Pull request by SerialVelocity**
Fixes #905

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/926)

### CODE REVIEW

File: atlasdb-ete-tests/build.gradle [hunk 1]
```diff
@@ -20,7 +20,6 @@ dependencies {
     compile project(':atlasdb-rocksdb')
 
     compile "io.dropwizard:dropwizard-core:" + libVersions.dropwizard
-    compile 'com.google.dagger:dagger:' + libVersions.dagger
```
Comment (gbrova) [inline atlasdb-ete-tests/build.gradle]: Nice

Comment (SerialVelocity) [inline atlasdb-ete-tests/build.gradle]: We'll soon find out if it works xD

**Comment by gbrova** [comment]
After testing against the internal product, I think this might reintroduce the issue fixed in #878. Will investigate more next week.

**Comment by gbrova** [comment]
Tested after the changes, everything seems happy now.

File: gradle/versions.gradle [hunk 2]
```diff
@@ -17,7 +17,7 @@ ext.libVersions =
     http_remoting: '0.13.0',
     joda_time: '2.7',
     snappy: '1.1.1.7',
-    dagger: '2.4',
+    dagger: '2.0.2',
```
Comment (jboreiko) [inline gradle/versions.gradle]: Might be worth putting a note here about a problem we ran into upgrading.

**Pull request merged**